src/workerd/server/workerd.capnp

```diff
@@ -356,6 +356,15 @@ struct Worker {
       # workerd will forward AnalyticsEngineEvents to designated service in the body of HTTP requests
       # This binding is subject to change and requires the `--experimental` flag
 
+      hyperdrive :group {
+        designator @18 :ServiceDesignator;
+        database @19 :Text;
+        user @20 :Text;
+        password @21 :Text;
+      }
+      # A binding for Hyperdrive. Allows workers to use Hyperdrive caching & pooling for Postgres
```

Comment:
Might also want to add a "scheme" field here given that we will start supporting other DB types before too long. If we don't add it now, then it'll either break existing configs when we do or we'll have to forever default the "scheme" field to "postgres".